### PR TITLE
[AzureMonitorExporter] Integration of OpenTelemetry Resource Detectors

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
* Added `Resource` to traces, logs, and metrics with default configuration.
* Added package `OpenTelemetry.ResourceDetectors.Azure` to introduce OpenTelemetry Resource detectors `AppServiceResourceDetector` and `AzureVMResourceDetector`.

* Update OpenTelemetry dependencies
  - OpenTelemetry 1.5.1
  - OpenTelemetry.Extensions.Hosting 1.5.1
  - OpenTelemetry.Instrumentation.AspNetCore 1.5.1-beta.1
  - OpenTelemetry.Instrumentation.Http 1.5.1-beta.1
  - OpenTelemetry.Instrumentation.SqlClient 1.5.1-beta.1

**Sample Telemetry**

```
{"name":"Metric","time":"2023-07-25T19:29:12.7579120Z","iKey":"00000000-0000-0000-0000-APP_SETTING","tags":{"ai.cloud.role":"MyWebApp","ai.cloud.roleInstance":"DESKTOP","ai.internal.sdkVersion":"dotnet7.0.9:otel1.5.1:ext1.0.0-alpha.20230725.1"},"data":{"baseType":"MetricData","baseData":{"metrics":[{"name":"_OTELRESOURCE_","value":0}],"properties":{"service.name":"MyWebApp","cloud.provider":"azure","cloud.platform":"azure_app_service","telemetry.sdk.name":"opentelemetry","telemetry.sdk.language":"dotnet","telemetry.sdk.version":"1.5.1"},"ver":2}}}`
```

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md [hunk 1]
```diff
@@ -4,12 +4,25 @@
 
 ### Features Added
 
+* Added `Resource` to traces, logs, and metrics with default configuration.
+  ([#37837](https://github.com/Azure/azure-sdk-for-net/pull/37837))
+* Added package `OpenTelemetry.ResourceDetectors.Azure` to introduce OpenTelemetry Resource detectors `AppServiceResourceDetector` and `AzureVMResourceDetector`.
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:9]:
```suggestion
* Added resource detection for `Azure App Service` and `Azure VM` environment. 
```

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:9]: Do we need a readme update on how to override these?

**Review by vishweshbankwar — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md (hunk 1 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:9]: Yes, we should a follow up PR.

File: eng/Packages.Data.props [hunk 2]
```diff
@@ -122,13 +122,14 @@
 
   <ItemGroup Condition="$(MSBuildProjectName.StartsWith('Azure.Monitor.OpenTelemetry'))">
     <!-- OpenTelemetry dependency approved for Azure.Monitor.OpenTelemetry.Exporter package only -->
-    <PackageReference Update="OpenTelemetry" Version="1.5.0" />
+    <PackageReference Update="OpenTelemetry" Version="1.5.1" />
     <PackageReference Update="OpenTelemetry.Exporter.InMemory" Version="1.5.0" />
```
Comment (TimothyMothra) [inline eng/Packages.Data.props:126]: can you also upgrade the InMemoryExporter to 1.5.1?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md [hunk 3]
```diff
@@ -4,12 +4,25 @@
...
 
 ### Other Changes
 
+* Update OpenTelemetry dependencies
+  ([#37837](https://github.com/Azure/azure-sdk-for-net/pull/37837))
+  - OpenTelemetry 1.5.1
+  - OpenTelemetry.Extensions.Hosting 1.5.1
+  - OpenTelemetry.Instrumentation.AspNetCore 1.5.1-beta.1
+  - OpenTelemetry.Instrumentation.Http 1.5.1-beta.1
+  - OpenTelemetry.Instrumentation.SqlClient 1.5.1-beta.1
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:24]: should also include a similar message in the Exporter Changelog

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:24]: Only for OpenTelemetry package?

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:24]: yup. that's what we've been doing :)

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md (hunk 1 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md]: @TimothyMothra Could you please help here once this PR gets merged?

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md]: This is covered already in readme - https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore#adding-custom-resource

I tested and it works.

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md]: Should we document `builder.Services.AddOpenTelemetry().ConfigureResource(resourceBuilder => resourceBuilder.AddService("service-name"));` which is common for all 3 signals.

**Pull request merged**